### PR TITLE
Upgrade to storybook v8

### PR DESCRIPTION
**Pull request by avatus**
New branch which brings in the changes from https://github.com/gravitational/teleport/pull/37331 . It was easier to do it this way over a rebase since the OG pr is a bit out of date

enterprise: https://github.com/gravitational/teleport.e/pull/3256

Closes gravitational/teleport-private#1494

### CODE REVIEW

**Comment by avatus** [comment]
I still need to resolve this error (which makes no sense to me yet)
```
 FAIL  web/packages/teleport/src/components/UserMenuNav/UserMenuNav.test.tsx
  ● Test suite failed to run

    ReferenceError: Response is not defined

      21 |
      22 | import { setupServer } from 'msw/node';
    > 23 | import { http, HttpResponse } from 'msw';
```

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (ravicious) [inline web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:109]: `PollingSuccess` was not the only story that was supposed to use `res.once`. AFAIK, at the time of the upgrade to v7, all of the uses of `res.once` in teleport.e were unnecessary (see https://github.com/gravitational/teleport.e/pull/3256/commits/5d0367d6cceedc71a62f7fe425873820ca6df1c0), but there was a couple of places in OSS that actually do needed it.

Could you go through them on master and add `{ once: true }` where necessary? I found that just going to the `web` dir and grepping for `res.once` is good enough.

File: jest.config.js
Comment (ravicious) [inline jest.config.js:34]: What is this for?

File: web/packages/teleport/src/lib/global.d.ts
Comment (ravicious) [inline web/packages/teleport/src/lib/global.d.ts:23]: We might not need this whole `declare global` declaration at all.

File: yarn.lock
Comment (ravicious) [inline yarn.lock:1]: `npx yarn-deduplicate yarn.lock` still seems to remove a couple of packages.

**Review by ravicious — COMMENTED** [review]
Geez, that's a ton of work with those `msw` changes. 🥵 Thanks a ton for doing this.

File: docs/check-redirects/check-redirects.test.js
Comment (ravicious) [inline docs/check-redirects/check-redirects.test.js:2]: I was wondering why the vercel bot keeps spamming under this PR. It turns out it's because of this change.

File: jest.config.js
Comment (avatus) [reply jest.config.js:34]: it had to do with the `setupServer` from msw but, I never got that working anyway. I'll fiddle around with those tests some more and if this is needed, I'll add a comment

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (avatus) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:109]: it doesn't work because the request itself is broken. they are returning 404s so its like... msw is and isn't intercepting all at the same time.

For some reason, none of my `msw` requests are working now (seems like only `enterprise` urls are working). I'm not sure what changed but I'll look into it. 

It was working during my fix so it must be related to something else. I'll keep digging around

Comment (avatus) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:109]: Ok so, if msw fails to fetch a URL (like it does everytime i go to polling success), then it breaks for everything else and nothing will work.

I've been fiddling with it for a few hours and still not sure the cause but this is a great story to look at as its the one that breaks.

**Comment by avatus** [comment]
I think I'll need another set of eyes on this `msw` change. something isn't quite right. I'll return to it on monday

File: web/.storybook/main.ts
Comment (gzdunek) [inline web/.storybook/main.ts:54]: I don't think using Web UI vite config is a good idea. It has a lot of stuff specific to Web UI that we have to disable via env vars or filtering out plugins. Actually, the only thing that we actually need from the Web UI config is `tsconfigPathPlugin` . 
Please take a look at the following simplified config:

```
Subject: [PATCH] Simplify storybook config
---
Index: package.json
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/package.json b/package.json
--- a/package.json	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/package.json	(date 1721047194393)
@@ -11,8 +11,8 @@
     "build-term": "yarn workspace @gravitational/teleterm build",
     "start-term": "yarn workspace @gravitational/teleterm start",
     "package-term": "yarn workspace @gravitational/teleterm package",
-    "storybook": "VITE_NO_PROXY=1 storybook dev -p 9002 -c web/.storybook",
-    "storybook-smoke-test": "VITE_NO_PROXY=1 yarn storybook --ci --smoke-test",
+    "storybook": "storybook dev -p 9002 -c web/.storybook",
+    "storybook-smoke-test": "yarn storybook --ci --smoke-test",
     "test": "jest",
     "test-coverage": "jest --coverage && web/scripts/print-coverage-link.sh",
     "test-update-snapshot": "yarn test --updateSnapshot",
@@ -105,5 +105,6 @@
     "workerDirectory": [
       "web/.storybook/public"
     ]
-  }
-}
\ No newline at end of file
+  },
+  "packageManager": "yarn@1.22.22+sha512.a6b2f7906b721bba3d67d4aff083df04dad64c399707841b7acf00f6b133b7ac24255f2652fa22ae3534329dc6180534e98d17432037ff6fd140556e2bb3137e"
+}
Index: web/packages/build/.babelrc.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/packages/build/.babelrc.js b/web/packages/build/.babelrc.js
--- a/web/packages/build/.babelrc.js	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/web/packages/build/.babelrc.js	(date 1721047036622)
@@ -41,6 +41,7 @@
       },
     ],
     ['@babel/preset-react', { runtime: 'automatic' }],
+    '@babel/preset-typescript',
   ];
 }
 
Index: web/.storybook/vite.config.mts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/.storybook/vite.config.mts b/web/.storybook/vite.config.mts
new file mode 100644
--- /dev/null	(date 1721047179104)
+++ b/web/.storybook/vite.config.mts	(date 1721047179104)
@@ -0,0 +1,9 @@
+import { defineConfig } from 'vite';
+import { tsconfigPathsPlugin } from '@gravitational/build/vite/tsconfigPaths.mjs';
+
+export default defineConfig(() => ({
+  // We don't use `reactPlugin` (which imports @vitejs/plugin-react-swc under the hood)
+  // because Storybook doesn't support SWC compilter in Vite-based projects.
+  // Once it's available, migrate from babel.
+  plugins: [tsconfigPathsPlugin()],
+}));
Index: web/.storybook/main.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/web/.storybook/main.ts b/web/.storybook/main.ts
--- a/web/.storybook/main.ts	(revision 6074d7df3b44dc64d5cbb39913d064385c2fc7ae)
+++ b/web/.storybook/main.ts	(date 1721047160779)
@@ -16,19 +16,15 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import path, { dirname, join, resolve } from 'path';
-import { PluginOption, UserConfig, Plugin } from 'vite';
+import fs from 'node:fs';
+import path from 'node:path';
+
 import type { StorybookConfig } from '@storybook/react-vite';
-import fs from 'fs';
 
 const enterpriseTeleportExists = fs.existsSync(
   path.join(__dirname, '/../../e/web')
 );
 
-function getAbsolutePath(value: string): any {
-  return dirname(require.resolve(join(value, 'package.json')));
-}
-
 function createStoriesPaths() {
   const stories = ['../packages/**/*.story.@(ts|tsx)'];
 
@@ -40,38 +36,13 @@
   return stories;
 }
 
-const rootDirectory = path.resolve(__dirname, '..', '..');
-const webDirectory = path.resolve(rootDirectory, 'web');
-
 const config: StorybookConfig = {
   stories: createStoriesPaths(),
   framework: {
-    name: getAbsolutePath('@storybook/react-vite'),
-    options: {
-      builder: {
-        viteConfigPath: resolve(
-          __dirname,
-          '../../web/packages/teleport/vite.config.mts'
-        ),
-      },
-    },
+    name: '@storybook/react-vite',
+    options: { builder: { viteConfigPath: 'web/.storybook/vite.config.mts' } },
   },
   staticDirs: ['public'],
-  async viteFinal(config: UserConfig) {
-    config.plugins = config.plugins?.filter((plugin: PluginOption) => {
-      if (plugin && 'name' in plugin) {
-        return (
-          plugin.name !== 'teleport-html-plugin' &&
-          plugin.name !== 'teleport-transform-html-plugin' &&
-          // TODO (avatus) migrate fully to the new react-docgen
-          // see migration guide from 6 -> 8 https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-docgen-component-analysis-by-default
-          plugin.name !== 'storybook:react-docgen-plugin'
-        );
-      }
-    });
-
-    return config;
-  },
 };
 
 export default config;
```

File: web/.storybook/main.ts
Comment (gzdunek) [reply web/.storybook/main.ts:68]: We don't have to disable this plugin. It didn't work because Storybook still uses babel for the Vite framework and our babel config wasn't fully configured.
After I added typescript preset to it, the errors gone away (I have it in the patch above).

File: web/.storybook/main.ts
Comment (ravicious) [reply web/.storybook/main.ts:54]: @gzdunek This patch was missing an empty line before the three backticks, I edited it to include that. Otherwise `git apply` fails with `error: corrupt patch at line 138`.

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:109]: When debugging this story, I found that the failing requests that msw complains about belong to some completely different stories. This made me realize that we have multiple stories which manually provide `mswLoader` and call `initialize`. I believe this has to be removed before we venture any further. It might even resolve the issue itself, as otherwise I don't see any problems with the config (I was testing things with Grzegorz's patch applied).

**Comment by avatus** [comment]
Applied and patch and removed the extra loaders/initialize callsites. the rest of the stories are much smoother/work more reliably but the SetupConnect/PollingSuccess still breaks horrendously and then any story used after is broke. Diving in more

**Comment by avatus** [comment]
I still need to fix the `UserContext.test`, i've been trying to [follow some docs ](https://mswjs.io/docs/migrations/1.x-to-2.x/#requestresponsetextencoder-is-not-defined-jest) to get `Response` defined but hit many many many more test failures while trying. I resorted to just removing `msw` and using `jest` to mock these response calls when needed. The last one i need to remove is from `UserContext.test` but it's eaten half the day so I'll return.

`SetupConnect` polling success still broke. I'll dive back in tomorrow

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: @ravicious can you check again? 

i noticed that some components who calls api from `teleportContext.<someService>.<apiCall>`, using the msw doesn't work, and i'm not sure why.


> This made me realize that we have multiple stories which manually provide mswLoader and call initialize

also i wasn't sure how i can tell this 🤔

File: package.json
Comment (kimlisa) [inline package.json:45]: this was required to make the theme switcher work again

File: web/.storybook/main.ts
Comment (kimlisa) [inline web/.storybook/main.ts:29]: we also have really old files that end in `jsx`, i put `js` just in case?

**Comment by kimlisa** [comment]
i clicked through every story and fixed _almost_ everything (EXCEPT teleterm... i couldn't handle it anymore, so i'll leave those to rafal or gz): see https://github.com/gravitational/teleport/pull/44041/commits/fa6fb1bc51d0db402815bd4f10d439b477b7e1a7

- I removed `main.story` because that's been broken for more than a year
- I couldn't fix `Player.story.tsx` because i think we need to somehow mock websocket
-  I can NOT figure out why [this and whereever else it is used](https://github.com/gravitational/teleport/blob/a567a74123494edbd2da1ade67f203d0789c98ad/web/packages/design/src/CardTerminal/CardTerminal.story.jsx) doesn't work... i don't get it and i don't understand what the story is doing 😅 (it doesn't work in master either). i tried putting in required fields but it doesn't work O_o

Here are some things i noticed:

- an api path with query params (with a question mark), makes stories break (and it starts breaking other stories). i found this [issue](https://github.com/mswjs/msw/issues/71) which i took it to mean it's not supported? i worked around this by stripping the affected path anything after the `?`
- msw doesn't seem to work with components that call api through the `teleportContext`: see https://github.com/gravitational/teleport/pull/44041#discussion_r1678813769, workaround is to mock the function through the context
  - mocking through context doesn't give us greatest flexibility (like how do we test pending states?), so another alternative is to not call api from contexts, we can do that later if no one cares

File: web/.storybook/vite.config.mts
Comment (kimlisa) [inline web/.storybook/vite.config.mts:9]: this was required for the styled-component `css` attribute to work

Comment (bl-nero) [reply web/.storybook/vite.config.mts:9]: Hmm, the comment says we _don't_ use `reactPlugin`, yet it's clearly used here. What don't I understand?

(Also: would you mind putting this statement directly into the code comment?)

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.story.tsx
Comment (ravicious) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.story.tsx:1]: I get this when I open `http://localhost:9002/?path=/story/teleport-discover-connectmycomputer-setupconnect--polling` (which is a completely different story).

```
[MSW] Uncaught exception in the request handler for "GET http://localhost:9002/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.story.tsx":

(bunch of indecipherable stacktrace)

This exception has been gracefully handled as a 500 response, however, it's strongly recommended to resolve this error, as it indicates a mistake in your code. If you wish to mock an error response, please see this guide: https://mswjs.io/docs/recipes/mocking-error-responses
```

This seems like Vite failing when Storybook asks for that file. @gzdunek I believe not importing the general vite config might be a good idea, but it looks like we're missing something still.

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: > > This made me realize that we have multiple stories which manually provide mswLoader and call initialize
> 
> also i wasn't sure how i can tell this 🤔

Michael already remove that from the stories, so there shouldn't be any that do this. Here's [an example of how this looked like](https://github.com/gravitational/teleport/pull/44041/commits/9122c178de28e38ce496ce7cd116edfc9931124e#diff-a6b09f9c6490139b7055114887d35ae202681837b28899d592700c426ce31134).

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx
Comment (ravicious) [inline web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx:43]: > an api path with query params (with a question mark), makes stories break (and it starts breaking other stories). i found this [issue](https://github.com/mswjs/msw/issues/71) which i took it to mean it's not supported (how did it work before)? i worked around this by stripping the affected path anything after the `?`

It might have been something related to msw 1.x, I vaguely remember getting this error in v7 update but I'm not sure.

Currently, I see this when I open `http://localhost:9002/?path=/story/teleporte-accesslists-list--list-unlimited`.

> [MSW] Found a redundant usage of query parameters in the request handler URL for "GET /v1/webapi/roles?startKey=&search=&limit=". Please match against a path instead and access query parameters using "new URL(request.url).searchParams" instead. Learn more: https://mswjs.io/docs/recipes/query-parameters

It comes from [`e/web/teleport/src/AccessListManagement/CreateAccessList/CreateAccessList.story.tsx`](https://github.com/gravitational/teleport.e/blob/ba1271c6faff653308e03292dcaacef918c8a1be/web/teleport/src/AccessListManagement/CreateAccessList/CreateAccessList.story.tsx#L66-L68).

I believe we should figure out why it was working, why it got broken, and how we're going to address this. Manually splitting the path in each story might not be the best way long term.

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:89]: The previous version works just fine after removing the query params from path and adding `{ once: true }`.

<details>
<summary>Code</summary>

```typescript
const noNodesHandler = http.get(cfg.api.nodesPath, () =>
  HttpResponse.json({ items: [] })
);

export const Polling = () => {
  return (
    <Provider>
      <SetupConnect {...setupConnectProps} />
    </Provider>
  );
};

Polling.parameters = {
  msw: {
    handlers: [noNodesHandler],
  },
};

export const PollingSuccess = () => {
  return (
    <Provider>
      <SetupConnect {...setupConnectProps} pingInterval={5} />
    </Provider>
  );
};

PollingSuccess.parameters = {
  msw: {
    handlers: [
      http.get(
        cfg.api.nodesPath.split('?')[0],
        () => {
          return HttpResponse.json({ items: [] });
        },
        { once: true }
      ),
      http.get(cfg.api.nodesPath.split('?')[0], () => {
        return HttpResponse.json({ items: [{ id: '1234', hostname: 'foo' }] });
      }),
    ],
  },
};

```
</details>

File: web/.storybook/preview.tsx
Comment (ravicious) [reply web/.storybook/preview.tsx:36]: After removing this arg, clearing the cache for `localhost` in my browser and opening `http://localhost:9002/?path=/story/teleport-discover-connectmycomputer-setupconnect--mac-os` for the first time, I see a bunch of warnings like this one:

```
[MSW] Warning: intercepted a request without a matching request handler:

  • GET /packages/teleport/src/Discover/yamlTemplates/awsAppAccessRO.yaml?import&raw

If you still wish to intercept this unhandled request, please create a request handler for it.
Read more: https://mswjs.io/docs/getting-started/mocks
```

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx:89]: @kimlisa Could you bring back the previous versions of those calls? I believe we can keep using msw instead of overriding functions if we remove the query params.

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx
Comment (ravicious) [reply web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx:43]: Perhaps it'd be possible to override `cfg.api` within storybook so that it returns some kind of a proxy object with a custom getter that always strips the query params?

Maybe there's a simpler way to replace the whole `cfg.api` in storybook with a version that doesn't use query params. We probably don't want to touch `teleport/src/config.ts` itself that much.

File: web/.storybook/vite.config.mts
Comment (gzdunek) [reply web/.storybook/vite.config.mts:9]: I'm not sure why this happens. Looks like `babel-plugin-styled-components` is not transforming the css prop (but I believe it should). I couldn't find anything useful in the storybook docs and issues, but I will keeping looking.

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: done: https://github.com/gravitational/teleport/pull/44041/commits/c8bc4fc7c201abb415f2098cdac873907b7c6891

Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: Ah sorry, I didn't mean only that story. In abb94e4d30355, there's a bunch of other stories that got those style of mocks, like `web/packages/teleport/src/Discover/Database/DeployService/ManualDeploy/ManualDeploy.story.tsx` or `web/packages/teleport/src/Discover/Kubernetes/HelmChart/HelmChart.story.tsx` – couldn't those be replaced with msw as well?

File: web/.storybook/preview.tsx
Comment (avatus) [reply web/.storybook/preview.tsx:36]: the `onUnhandledRequest` existed previously in our current config which is why I left it in here. Perhaps we can handle it more elegantly now but was just keeping up with previous work. will look into it

File: web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx
Comment (avatus) [reply web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx:43]: I believe we can setup something that will remove query params. i saw it somewhere in the msw docs, I'll look around again today

Comment (ryanclark) [reply web/packages/teleport/src/Discover/Database/EnrollRdsDatabase/EnrollRdsDatabaseEnroll.story.tsx:43]: It would be nice if the query params weren't in the cfg, it's one of the things blocking the upgrade to react-router 6 😔

Completely out of scope for this PR though of course

**Comment by avatus** [comment]
I didn't get to try removing the query params today but will tomorrow.

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: right, sorry, those were probably after i realized that query was the issue 😅 , i'll get to those tomorrow

**Comment by avatus** [comment]
Removing the query param made most of the stories (`SetupConnect` especially) work right away. I abstracted it to a function and into a new file that we might be able to add any other utils for storybook to (but i didnt call it utils.ts 😏 )

**Comment by avatus** [comment]
I'll rebase this onto the pnpm master today

**Comment by avatus** [comment]
In general, stories seem to work now. stories with `msw` work and if they don't can be fixed with my proposed `withoutQuery` util function. I'll update them if we agree on it. 

I disabled one test because jest gets rid of a bunch of globals that `msw` was failing with when using. I added a TODO to fix it, but I don't want one test to block this from being merged.

File: web/packages/teleport/src/User/UserContext.test.tsx
Comment (kimlisa) [inline web/packages/teleport/src/User/UserContext.test.tsx:20]: documenting [here](https://stackoverflow.com/questions/69881929/msw-with-jest-confusion)

**Review by kimlisa — APPROVED** [review]
:shipit:

File: web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Discover/ConnectMyComputer/SetupConnect/SetupConnect.story.tsx]: @ravicious https://github.com/gravitational/teleport/pull/44041/commits/619fdf2a94661b2ad68f6f540ddc43bb19ff8515, places where we used both msw and context mocking, i replaced context mocking with msw. stories that just used context mock i kept it the same

File: web/.storybook/vite.config.mts
Comment (kimlisa) [reply web/.storybook/vite.config.mts:6]: i don't understand it either, i just removed the comment for now

File: web/packages/build/storybook.ts
Comment (ryanclark) [inline web/packages/build/storybook.ts:3]: This shouldn't have been changed

**Review by ryanclark — APPROVED** [review]

File: web/.storybook/vite.config.mts
Comment (gzdunek) [reply web/.storybook/vite.config.mts]: I gave up on that, I don't know why it doesn't work with the babel plugin alone. I only corrected the mode passed to the react plugin.

File: web/packages/teleport/src/Notifications/Notification.story.tsx
Comment (ravicious) [inline web/packages/teleport/src/Notifications/Notification.story.tsx:137]: Here `res.once` was necessary, as there's another handler below for `cfg.api.notificationsPath`.

File: web/packages/teleport/src/mocks/handlers.ts
Comment (ravicious) [inline web/packages/teleport/src/mocks/handlers.ts:25]: These handlers don't seem to be used anywhere. Grepping the project for `handlersTeleport` returns only this file.

File: jest.config.js
Comment (ravicious) [reply jest.config.js:41]: Do we still need it?

File: web/packages/build/package.json
Comment (avatus) [reply web/packages/build/package.json:50]: it's used in an external script but perhaps it can be removed from here

File: web/packages/teleport/src/Sessions/Sessions.story.tsx
Comment (kimlisa) [reply web/packages/teleport/src/Sessions/Sessions.story.tsx:76]: it was used, but i can see how it was confusing, i removed in favor of just calling the func each time

File: web/packages/build/.babelrc.js
Comment (avatus) [reply web/packages/build/.babelrc.js:44]: This is for the react-docgen in the new storybook upgrade. without it, it fails.

File: package.json
Comment (avatus) [reply package.json:60]: it needs to be in root and NOT build. in build only (after fresh `pnpm install`) causes storybook to not be found. i removed from build and kept it here

File: jest.config.js
Comment (avatus) [reply jest.config.js:41]: yes we do still need it for `setupServer` in our tests. More info here in migration guide
https://mswjs.io/docs/migrations/1.x-to-2.x/#cannot-find-module-mswnode-jsdom

**Review by ravicious — APPROVED** [review]
🚀

File: package.json
Comment (ravicious) [inline package.json:39]: I'm not sure if this needs to be here. 🤔

Comment (avatus) [reply package.json:39]: the vite config filein storybook wasn't able to find this package without having this here. i could use relative paths but i wanted to keep the import cleaner

**Pull request merged**
